Tribler/community/multichain/conversion.py

```diff
@@ -0,0 +1,108 @@
+"""
+File contains all conversions for the MultiChain Community.
+"""
+from struct import pack, unpack_from, calcsize
+
+from Tribler.dispersy.conversion import BinaryConversion
+from Tribler.dispersy.message import DropPacket
+
+"""
+Hashlength used in the MultiChain Community
+"""
+HASH_LENGTH = 20
+"""
+Formating of the signature packet
+"""
+request_format = '!I I I I i ' + str(HASH_LENGTH) + 's'
+signature_format = request_format + ' i ' + str(20) + 's'
+signature_size = calcsize(signature_format)
+request_size = calcsize(request_format)
+
+
+class MultiChainConversion(BinaryConversion):
+    """
+    Class that handles all encoding and decoding of MultiChain messages.
+    """
+
+    def __init__(self, community):
+        super(MultiChainConversion, self).__init__(community, "\x01")
+        from Tribler.community.multichain.community import SIGNATURE
+
+        # Define Request Signature.
+        self.define_meta_message(chr(1), community.get_meta_message(SIGNATURE),
+                                 self._encode_signature, self._decode_signature)
+
+    def _encode_signature(self, message):
+        """
+        Encode the signature message
+        :param message: Mesage.impl of SIGNATURE
+        :return: encoding ready to be sent of the network.
+        """
+        payload = message.payload
+        return pack(signature_format, *(payload.up, payload.down, payload.total_up, payload.total_down,
+                                        payload.sequence_number_requester, payload.previous_hash_requester,
+                                        payload.sequence_number_responder, payload.previous_hash_responder)),
+
+    def _decode_signature(self, placeholder, offset, data):
+        """
+        Decode an incoming signature message
+        :param placeholder:
+        :param offset: Start of the SIGNATURE message in the data.
+        :param data: ByteStream containing the message.
+        :return: (offset, SIGNATURE.impl)
+        """
+        if len(data) < offset + signature_size:
+            raise DropPacket("Unable to decode the payload")
+
+        values = unpack_from(signature_format, data, offset)
+        offset += signature_size
+
+        if len(values) != 8:
+            raise DropPacket("Unable to decode the signature")
+
+        up = values[0]
+        if not isinstance(up, int):
```

Comment:
No need to check the types here, struct will make an int due to the signature_format